### DISCUSSION TITLE
Default styling options on ScatterplotLayer

### DISCUSSION THREAD

**Discussion by zacdezgeo** [developmentseed/lonboard#577] [Ideas]
Hey @kylebarron! I'm not sure if this is the right spot for this discussion. Please point me to the right path as needed.

So, I've been using lonboard with a hexagon dataset. I wanted to do more advanced styling, but it took me a moment to debug how to use the Scatterplotlayer. I thought I was encountering some issue because the points were not showing up on the map:
![image](https://github.com/user-attachments/assets/d6f3e873-393c-4653-83ca-68596a157518)

But the `viz()` function was rendering my `gdf` without any issues:
![image](https://github.com/user-attachments/assets/eea064b4-2e01-4308-8c95-74246f635fa7)

Turns out that the default radius was set to `None`:
![image](https://github.com/user-attachments/assets/046d59e7-0a16-4951-8b07-a3571c946743)

It seems `viz()` has some defaults where it sets the radius and the random color, while the `ScatterplotLayer` does not.
Is this the expected behavior? There may be some reason I'm unaware of, but I would've expected the `ScatterplotLayer` to have the same default settings as the `viz()` function.

**Comment by kylebarron** [comment]
The question of what to use for defaults is indeed tricky.

Upstream, deck.gl has default values for virtually all parameters. So for example, if you look at the [JS ScatterplotLayer docs](https://deck.gl/docs/api-reference/layers/scatterplot-layer), you'll see all the render options and data accessors. And the [default for `get_radius` is 1](https://deck.gl/docs/api-reference/layers/scatterplot-layer#getradius). That, combined with the [default for `radiusUnits` being "meters"](https://deck.gl/docs/api-reference/layers/scatterplot-layer#radiusunits) means that by default, deck.gl renders a `ScatterplotLayer` with points that are 1 meter wide. Of course, this default means that points are almost never visible without the user passing explicit parameters. (Note that in Lonboard `None` does **not** mean "invisible". Rather it means "defer to deck.gl's default value in JavaScript")

This also speaks to differences between the audience of the JS library and the Python library. Users of the JS library are usually building an application, where they use a single dataset or class of data, and spend a lot of time making it look good. Lonboard, on the contrary, is more often used for exploratory data analysis, where you might have lots of different datasets you want to inspect, and where just seeing your data is more important than exactly how it looks.

This is the rationale between having `lonboard.viz` and `lonboard.Layer` have different defaults: use `viz` for exploratory data analysis where you _just want to see your data_, and use the latter for customizing exactly what you want to see.

Maybe Lonboard's layers should have defaults? But choosing a good default is really hard, and choosing _when to apply those defaults_ is even harder. If a user passes in `get_radius`, should we rescind any defaults for `radius_min_pixels` and `radius_max_pixels`?

**Comment by zacdezgeo** [comment]
I'm not sure whether the minimum or maximum radius should be used; either would probably work. Still, for users onboarding a lonboard, seeing their data without immediately diving into the intricacies of styling would make a difference. The benefit to user onboarding outweighs the tradeoff of potentially displaying the data in a style that is not quite proper.

**Reply by kylebarron** [reply]
But the question of _when_ to apply defaults is still really hard. If you pass

```py
ScatterplotLayer.from_geopandas(gdf, get_radius=...)
```
should any defaults be set? Because the default unit for radius is meters, so if you pass `get_radius=1`, you won't see anything on the map. But alternatively, if we set a default of `radius_min_pixels=5` or something, then the user would have to turn _off_ the default when they actually want small points.

**Reply by kylebarron** [reply]
Maybe this is well-served by having a `default_styling: bool` parameter, where the user can turn on/off whether defaults should be applied. But then the question of: what should be the default value of `default_styling`?

**Reply by zacdezgeo** [reply]
Not sure there's a right answer. IMHO I feel that a user who wants to customize their styling will be less burdened by setting these parameters than users onboarding and not seeing their points.

**Reply by kylebarron** [reply]
In #620 we added an `add_layer` method to `Map`, so now it's easier to add another point layer with default styling settings with `map.add_layer(viz(gdf_with_point_data))`. This makes me more inclined to point people who want default styling settings to use `viz`, and potentially change the styling attributes after the fact:
```py
m = viz(point_data)
layer = m.layers[0]
layer.get_radius = 5
```
